### DISCUSSION TITLE
How to enable writing in input fields ?

### DISCUSSION THREAD

**Discussion by avanisingh0105** [express-labs/pure-react-carousel#367] [Q&A]
I am working on a functionality of a multi-step form with pure-react-carousel. I am using custom components (which are forms) as slides but not able to type anything in the input field, the input field is not getting focused on clicking them in any of the slides. Can someone please help me, how can I enable typing in the input fields?

**Comment by Conor-Hinchee** [comment]
Hello @avanisingh0105 !

I think you are going to have a problem using our carousel for the multi-step form you are building. The project wasn't built with this functionality as a goal and the main concerns I have would be how are `Slide` component captures the user events to determine if the carousel should move to the next slide or not. You can pass a custom `onClick` handler to the `Slide` which can focus the inputs in the form using references but as the `Slide` wraps the entire form,but right now we are not able to determine what exactly a user is attempting to click. 

Here is an example of how it might work but I am not sure if this project will work well as a solution for you.
https://codesandbox.io/s/cranky-margulis-nij8j?file=/src/App.js

**Reply by avanisingh0105** [reply]
Hi! 
Thanks for helping. This codesandbox might work as a solution but for now I have moved on from pure-react-carousel as a dependency for my project.
